### ISSUE TITLE
`PNPM_HOME` does not expand on Windows

### ISSUE THREAD

**Issue by maxichrome** [pnpm/pnpm#5696]
### pnpm version:

7.17.0

### Code to reproduce the issue:

_n/a_

### Expected behavior:

`%PNPM_HOME%` expands when inserted into other environment variables such as `Path`.

#### Example

If Path is set to `%PNPM_HOME%;%USERPROFILE%\.bin`, when opening a shell it becomes `C:\Users\user\AppData\Local\pnpm;C:\Users\user\.bin`

#### Example

If Path is set to `%PNPM_HOME%;%USERPROFILE%\.bin`, when opening a shell it becomes `%PNPM_HOME%;C:\Users\user\.bin`

### Additional information:

 - `node -v` prints: v19.1.0
 - Windows, macOS, or Linux?: Windows

#### Suspected cause

This has been reproduced by myself and @maplefins in further testing:

During `pnpm setup`, pnpm sets an environment variable called `PNPM_HOME`. On Windows, it does this by directly calling **reg.exe** instead of using a native system API (see [`path-extender-windows.ts`](https://github.com/pnpm/components/blob/7b8f9eed34fd33c988ad54679a2d747182d3c062/os/env/path-extender-windows/path-extender-windows.ts#L146); also see Workaround below). 

#### Reproduced on

 - Windows Server 2022 [20348.1311]
 - Windows 11 [22621.898]

#### Workaround

Update the `PNPM_HOME` variable via a native API (in a shell or the Environment Variables editor UI).

In PowerShell:
     
```powershell
$pnpmHome = [System.Environment]::GetEnvironmentVariable("PNPM_HOME", [System.EnvironmentVariableTarget]::User)
[System.Environment]::SetEnvironmentVariable("PNPM_HOME", $pnpmHome, [System.EnvironmentVariableTarget]::User)
```

**Comment by zkochan** [comment]
How can we run then native API? Should we generate a powershell file and run it? As far as I remember Windows doesn't allow to run unsigned ps1 files.

**Comment by maxichrome** [comment]
I'm unsure what this looks like practically in terms of security-related permissions but you _should_ be able to run the PowerShell executable with the necessary function as an argument

**Comment by maxichrome** [comment]
Based on the fact this works:

```powershell
powershell -command "[System.Environment]::SetEnvironmentVariable('PNPM_HOME', $pnpmHome, [System.EnvironmentVariableTarget]::User)
```

`path-extender-windows.ts` could be modified in a way similar to this:

```typescript
await execa('powershell', ['-command', '"[System.Environment]::SetEnvironmentVariable(\'${envVarName}\', \'${envVarValue}\', [System.EnvironmentVariableTarget]::User)"'], EXEC_OPTS)
```

You may want to split this up somehow to avoid the long single string, but that's all for however you want to style your own code :P

**Comment by zkochan** [comment]
By the way, for this issue there was this suggestion: https://github.com/pnpm/pnpm/issues/4658#issuecomment-1304421148

that I released in v7.17.1

That didn't help?

**Comment by zkochan** [comment]
Looks like it is solved but only if PNPM_HOME wasn't configured by pnpm older than v7.17

https://github.com/pnpm/pnpm/issues/4658#issuecomment-1404199772

**Comment by maxichrome** [comment]
yes the findings in that comment make sense with my previous findings! maybe a separate change needs to be made to retrieve, unset, and reset `PNPM_HOME` to the same value using the current method

(or would unsetting be necessary at all?)
